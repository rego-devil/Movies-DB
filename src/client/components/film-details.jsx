import React from 'react';
import { Link } from 'react-router-dom';
import Loading  from './loading';

export class FilmDetails extends React.Component {

  componentWillMount() {
    this.props.getFilm(this.props.match.params.id);
  }

  render() {
    const {currentFilmDetails, onClearCurrentFilm, isFetching} = this.props;    

    if(!currentFilmDetails) 
      return null

    return (
      <div className="filmDetails">
       
        <img className="filmDetails__img"  src={currentFilmDetails.poster_path} />
        <div className="filmDetails__overview">
          <div className="filmDetails__top">
            <div className="filmDetails__title">{currentFilmDetails.title}</div>
            <div className="filmDetails__rating">{currentFilmDetails.vote_average}</div>
            
              <Link to="/">
                <button className="filmDetails__backToSearch" onClick={() => onClearCurrentFilm(null)}>
                  Back to search
                  </button>
              </Link>
            
          </div>
          <div className="filmDetails__middle">
            <div className="filmDetails__date">{currentFilmDetails.release_date && currentFilmDetails.release_date.split('-')[0]}</div>
            <div className="filmDetails__duration">{currentFilmDetails.runtime ? currentFilmDetails.runtime + ` min.` : ''}</div>
          </div>
          <div className="filmDetails__description">{currentFilmDetails.overview}</div>
          <div className="filmDetails__genres">
          {
            currentFilmDetails.genres && currentFilmDetails.genres.map((item, i) => (i > 0 ? ', ':'') + item)
          }
          </div>
        </div>
      </div>
      
      
      // currentFilmDetails ? (
      //     result
      // ) : (isFetching ? 
      //   <React.Fragment>
      //     { console.log('----3----')}
      //     <Loading />
      //   </React.Fragment> : '')
        
    )
  }
  
}