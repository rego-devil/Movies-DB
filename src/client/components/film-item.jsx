import React from 'react';
import propTypes from 'prop-types';


export const FilmItem = ({title, genres, release_date, poster_path}) => {

	let genreList = genres.map((item,i) => {
		return (i > 0 ? ', ':'') + item;
	});

	return (
		<React.Fragment>
			<img src={poster_path} className="filmItem__img" />
			<div className="filmItem__description">
					<div className="filmItem__title">{title}</div>
					<div className="filmItem__releaseYear">{release_date.split('-')[0]}</div>
			</div>
			<div className="filmItem__genre">{genreList}</div>
		</React.Fragment>
	)
}

FilmItem.propTypes = {
	title: propTypes.string.isRequired,
	release_date: propTypes.string.isRequired,
	poster_path: propTypes.string.isRequired,
	genres: propTypes.array,
}