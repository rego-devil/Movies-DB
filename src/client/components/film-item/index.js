export { default } from './film-item';
