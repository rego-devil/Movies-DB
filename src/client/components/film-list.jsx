import React from 'react';
import {FilmItem, FilmSort} from './';

export class FilmList extends React.Component {
	constructor() {
		super();
	}

	render() {
		const {filmList, onShowFilmDetails, onSelectFilm} = this.props;		
		return (
			filmList.length > 0 ? (
				<React.Fragment>
				<FilmSort />
				<ul className="filmList">
					{
						filmList.map((item) => 
							<li key={item.id} className="filmItem" onClick={() =>  onShowFilmDetails(item) }>
								<FilmItem {...item} />
							</li>
						)
					}
				</ul>
				</React.Fragment>
			) : (
				<div className="noFilms">No films found</div>
			)
		)
	}
}


