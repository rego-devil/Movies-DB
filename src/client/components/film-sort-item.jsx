import React from 'react';


export const SortItem = ({sortBy, onSort, currentSortField, value, sortOrderLocal }) => (
	<span 
		className={`filmSort__item ${currentSortField === sortBy ? 'filmSort__item--active' : ''}`} 
		onClick={() => onSort({sortBy: currentSortField, sortOrder: sortOrderLocal}) } 
	>
		{value}
		{currentSortField === sortBy ? sortOrderLocal === 'asc' ? String.fromCharCode(8593) : String.fromCharCode(8595) : ''  }
	</span>
)