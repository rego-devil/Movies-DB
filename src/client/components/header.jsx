import React from 'react';
import {Search, FilmDetails} from './';

export class Header extends React.Component {

    render() {
        const {currentFilmDetails, onShowFilmDetails, onSearch, searchBy, onSearchBy} = this.props;
        console.log(this.props);
        return (
            <div className="header">
                <div className="wrapper">
                    {
                        currentFilmDetails ? 
                            <FilmDetails {...currentFilmDetails} onShowFilmDetails={onShowFilmDetails}   /> 
                        :
                            <Search onSearch={onSearch} searchBy={searchBy} onSearchBy={onSearchBy} />
                    }
                </div>
            </div>
        )
    }
    
}