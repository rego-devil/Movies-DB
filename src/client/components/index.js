export { Header } from './header';
export { FilmItem } from './film-item';
export { FilmList } from './film-list';
export { FilmDetails } from './film-details';
export { FilmSort } from './film-sort';
export { SortItem } from './film-sort-item';
export { Search } from './search';
export { Footer } from './footer';
