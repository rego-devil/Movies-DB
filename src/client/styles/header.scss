.header {
	// background: url('../images/back.jpg') no-repeat center bottom;
	background-color: rgb(36, 36, 36);
	// background-size: cover;
	color: #fff;
	padding: 0 0 10px 0;
	position: fixed;
	top: 0;
	z-index: 100;
	width: 100%;
	min-height: 200px;


	&__siteName {
		padding: 10px 0 0 0;
		color: red;
		display: flex;
		justify-content: space-between;
	}

}

.search {

	&__title {
		text-transform: uppercase;
		padding: 10px 0 10px 0;
	}

	&__input {
		height: 40px;
		padding: 0 18px 0 18px;
		display: block;
		background: #000;
		color: #fff;
		width: 100%;
		font-size: 1.5rem;
		font-family: Roboto, sans-serif;
		outline: none;
		resize: none;
		border: none;
		border-bottom: 1px solid rgb(255, 94, 94);
	}

	&__button {
		line-height: 38px;
		background: red;
		color: #fff;
		display: inline-block;
		text-align: center;
		cursor: pointer;
		outline: none;
		padding: 0 10px;
		font-size: 1rem;
		border: 1px solid transparent;
		white-space: nowrap;
		width: 200px;
		

		&:hover {
			background: rgb(255, 94, 94);
		}

		&:active {
			background: rgb(165, 40, 40);
		}
	}

	&__bottom {
		display: flex;
		justify-content: space-between;
		align-items: center;
		padding: 10px 0 0 0;
	}

	&__filter {
		display: flex;
		align-items: center;
	}

	&__filterText {
		text-transform: uppercase;
		color: #fff;
	}

	&__filterItem {
		background: #888888;
		display: inline-block;
		padding: 4px 10px;
		margin-left: 10px;
		cursor: pointer;
		text-transform: uppercase;

		&--active {
			background: red;
		}
	}
}