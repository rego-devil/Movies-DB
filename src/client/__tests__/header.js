import React from 'react';
import {shallow, mount, render} from 'enzyme';
import {Header} from '../components/header';

describe('Header  component', () => {
    let renderedComponent;

    const minProps = {
        showFilmDetails: true,
        onShowFilmDetails: () => {} 
    }

    beforeEach(()=>{
        renderedComponent = shallow(<Header {...minProps} />)
    });

    test('+++Match snapshot', () => {
        expect(renderedComponent).toMatchSnapshot();
    });

    test('renders without crashing', () => {
        expect(renderedComponent.length).toEqual(1);
    });

    test('render with various props"  ', () => {
        
        let renderedComponent = shallow(<Header currentFilmDetails={null} />);
        expect(renderedComponent.find('Search').length).toBe(1);


        renderedComponent = shallow(<Header currentFilmDetails={{}} />);
        expect(renderedComponent.find('FilmDetails').length).toBe(1);

        console.log(renderedComponent.debug());
    });

});