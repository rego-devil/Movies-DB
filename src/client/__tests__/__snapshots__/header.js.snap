// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header  component renders without crashing 1`] = `
<div
  className="header"
>
  <div
    className="wrapper"
  >
    <Search />
  </div>
</div>
`;
